### ISSUE TITLE
Convert images in a post from JPG to WEBP

### ISSUE THREAD

**Issue by tripsoverpoland** [Automattic/jetpack#34879]
### Impacted plugin

Jetpack

### What

When editing a post, there should be an option: convert images in this post from JPG to WEBP

### How

Many blogs with a history longer than a few years have collected large amounts of images in the old JPG format.
It would be nice if Jetpack allowed images in these older entries to be converted to the newer WEBP format.
This should be possible by adding some button called:
"Convert images from JPG to WEBP"
when editing older entries.
The conversion should apply to all images in the post inserted there in JPG format.
Please remember that all versions (sizes) of JPG images in the media archive have also been converted to the WEBP format for this entry.
Also, the html code of the image names in the entry must be changed so that *.JPG images are changed to *.WEBP in it.
What remains to be discussed is what to do with the original JPG images in the media archive - this may be a configuration option:
1)Move JPG images (before conversion) to another folder
or
2) add a suffix to the JPG image names (before conversion), e.g. image-webp.jpg.

**Comment by jeherve** [comment]
Creating new image files on your site is probably not something we'll implement in the Jetpack plugin. On sites with a lot of images, it could mean significantly increasing the disk space used by the media library.

Instead, I would recommend using Jetpack's Image CDN feature. The CDN automatically creates webP versions of all the images on your site, but those are cached and served from the CDN instead of being stored on your own server. To use that feature, go to Jetpack > Settings > Performance and turn on the site accelerator feature. From that point on, your site will serve webP images for all the images you've inserted in your posts, as long as you access the posts with a browser that supports the webP format.

**Comment by tripsoverpoland** [comment]
I understand the limited disk space on user hosting sites.
  However, it is up to the user whether to leave the *.JPG originals saved before conversion on their hosting or to copy them to their local archive and delete them from their hosting.
  The moment of conversion from *.JPG to *.WEBP is so critical on the user's local hosting that it may be interrupted at a random moment, for example, after converting a random number of images, and it may be difficult to complete or repeat it in its entirety.
  Of course, using a CDN is a solution, but it may also involve, for example, the temporary unavailability of the CDN for website/blog users.
  Therefore, from the point of view of the integrity of serving data and images from the blog hosting, it would be better to have all the data and images on the blog hosting, and only additionally have them on the CDN.
  Therefore, just like in order to change the Tiled Gallery into a regular Wordpress gallery (and vice versa), it is a good idea for the user to be able to replace *.JPG images in any entry with the *.WEBP format according to his/her intention,
  leaving the *.JPG originals on your server for a while to make sure that the conversion process took place without any interruptions and all images from the entry were converted correctly,
  and finally the html code of the entry was also changed, replacing *.JPG with *.WEBP.
  If there is no such possibility of changing image formats in the selected entry, the user, in addition to the mentioned CDN, can do manual conversion using tools outside the blog.
  I used another plugin that converted images from JPG to WEBP for some time, but unfortunately Jetpack did not work well with it, creating numerous html errors in the code,
  so a few weeks ago I gave up this plug-in, agreeing to leave only the *.JPG format in older entries, and in newer entries (since several years since Wordpress already supports *.WEBP) by inserting only *.WEBP images.
  However, this is not a 100% perfect solution, which is why I submitted this idea to Jetpack.
